File: laba2/lab2.cs
using System;

class Program {
    static int SearchColor(string r) {
        if (r.Length % 2 != 0) {
            Console.WriteLine("Incorrect value");
            return 0;
        }

        int c = 0;
        int[] VecR = new int[10];
        int[] VecW = new int[10];
        int[] VecB = new int[10];

        for (int i = 0; i < r.Length; i += 2) {
            char color = r[i];
            int cod = r[i + 1] - '0';

            if (color == 'R') VecR[cod] = 1;
            else if (color == 'W') VecW[cod] = 1;
            else if (color == 'B') VecB[cod] = 1;
        }

        for (int i = 0; i < 10; i++) {
            if (VecR[i] == 1 && VecW[i] == 1 && VecB[i] == 1) c++;
        }

        return c;
    }

    static void Main() {
        Console.Write("Enter rods: ");
        string a = Console.ReadLine();
        Console.WriteLine("Number of rods: " + SearchColor(a));
    }
}

File: laba2/lab2.rs
use std::io;

fn search_color(r: &str) -> i32 {
    if r.len() % 2 != 0 {
        println!("Incorrect value");
        return 0;
    }

    let mut vec_r = [0; 10];
    let mut vec_w = [0; 10];
    let mut vec_b = [0; 10];
    let chars: Vec<char> = r.chars().collect();

    for i in (0..r.len()).step_by(2) {
        let color = chars[i];
        let cod = chars[i + 1].to_digit(10).unwrap() as usize;

        match color {
            'R' => vec_r[cod] = 1,
            'W' => vec_w[cod] = 1,
            'B' => vec_b[cod] = 1,
            _ => {}
        }
    }

    let mut c = 0;
    for i in 0..10 {
        if vec_r[i] == 1 && vec_w[i] == 1 && vec_b[i] == 1 {
            c += 1;
        }
    }
    c
}

fn main() {
    let mut input = String::new();
    println!("Enter rods: ");
    io::stdin().read_line(&mut input).unwrap();
    let input = input.trim();
    println!("Number of rods: {}", search_color(input));
}
